generic-database-plugin/src/main/java/co/cask/jdbc/DatabaseSource.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.jdbc;
+
+import co.cask.ConnectionConfig;
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Macro;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.db.batch.source.DBSource;
+import org.apache.commons.lang.StringUtils;
+
+import javax.annotation.Nullable;
+
+/**
+ * Batch source to read from generic database.
+ */
+@Plugin(type = "batchsource")
+@Name(DatabaseConstants.PLUGIN_NAME)
+@Description("Reads from a database table(s) using a configurable SQL query." +
+  " Outputs one record for each row returned by the query.")
+public class DatabaseSource extends DBSource {
```

Comment:
Let's rename all the base classes (DBAction, DBSource, DBSink) to start with 'Abstract' (AbstractDBAction, AbstractDBSource, ...). Otherwise it can be a little confusing which one is the actual plugin and which one is the base class.

Reply:
done